File: 3154-maximum-value-of-an-ordered-triplet-i/3154-maximum-value-of-an-ordered-triplet-i.java
class Solution {
    public long maximumTripletValue(int[] nums) {
        long m = Long.MIN_VALUE;
        for(int i = 0; i < nums.length; i++){
            for(int j = i+1; j < nums.length; j++){
                for(int k = j+1; k < nums.length; k++){
                    m = Math.max(m, (long)(nums[i] - nums[j]) * nums[k]);
                }
            }
        }
        return Math.max(m, 0);
    }
}

File: 3154-maximum-value-of-an-ordered-triplet-i/3154-maximum-value-of-an-ordered-triplet-i.py
class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        m = float('-inf') 
        for i in range(0, len(nums)):
            for j in range(i+1, len(nums)):
                for k in range(j+1, len(nums)):
                     m = max(m, (nums[i] - nums[j]) * nums[k])
        
        return max(m, 0)